### ISSUE TITLE
VM failed to get write lock 

### ISSUE THREAD

**Issue by stefanDeveloper** [kubevirt/kubevirt#6906]
**What happened**:
I'm currently testing this [Windows Server example](https://kubevirt.io/2020/KubeVirt-installing_Microsoft_Windows_from_an_iso.html) to get familiar with KubeVirt and I receive the following message during startup: `kubectl describe vmi win2k12-iso`:

```
server error. command SyncVMI failed: "LibvirtError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 2021-12-02T14:48:39.812232Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.4,addr=0x0,drive=libvirt-1-format,id=ua-harddrive,write-cache=on,werror=stop,rerror=stop: Failed to get \"write\" lock\nIs another process using the image [/var/run/kubevirt-private/vmi-disks/harddrive/disk.img]?')"
```

I've also tested the following https://github.com/kubevirt/kubevirt/issues/6731#issuecomment-964888498, however it didn't help.

Any ideas why my VM runs into a lock problem?

**What you expected to happen**:

VM should start as expected

**How to reproduce it (as minimally and precisely as possible)**:

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: winhd
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 15Gi
  storageClassName: hostpath

apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: win2k12-iso
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/domain: win2k12-iso
    spec:
      domain:
        cpu:
          cores: 4
        devices:
          disks:
          - bootOrder: 1
            cdrom:
              bus: sata
            name: cdromiso
          - disk:
              bus: virtio
            name: harddrive
          - cdrom:
              bus: sata
            name: virtiocontainerdisk
        machine:
          type: q35
        resources:
          requests:
            memory: 8G
      volumes:
      - name: cdromiso
        persistentVolumeClaim:
          claimName: iso-win2k12
      - name: harddrive
        persistentVolumeClaim:
          claimName: winhd
      - containerDisk:
          image: kubevirt/virtio-container-disk
        name: virtiocontainerdisk
```

**Harddrive**

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: task-pv-volume-winhd
  labels:
    type: local
spec:
  storageClassName: hostpath
  capacity:
    storage: 15Gi
  accessModes:
    - ReadWriteMany
  hostPath:
    path: "/mnt/data"
```

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: winhd
spec:
  storageClassName: hostpath
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 15Gi
```

**Anything else we need to know?**:

cluster has been installed freshly

**Environment**:
- KubeVirt version (use `virtctl version`):
```sh
Client Version: version.Info{GitVersion:"v0.47.1", GitCommit:"c34de42a48f5564f4fd2c21b6cbda7b96664c65b", GitTreeState:"clean", BuildDate:"2021-11-11T16:01:45Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"v0.47.1-dirty", GitCommit:"c34de42a48f5564f4fd2c21b6cbda7b96664c65b", GitTreeState:"dirty", BuildDate:"2021-11-11T16:20:45Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes version (use `kubectl version`):
```sh
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-17T15:48:33Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-17T15:42:41Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
```
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
```sh
NAME="CentOS Linux"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
```
- Kernel (e.g. `uname -a`):
```sh
Linux kubernetes-master 4.18.0-348.2.1.el8_5.x86_64 #1 SMP Tue Nov 16 14:42:35 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Others:

**Comment by alicefr** [comment]
@stefanDeveloper could you please provide the virt-handler output?
You might want to try this work around. Instead of using the PVC, you could try to substitute with a blank datavolume:
```yaml
---
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: win2k12-iso
spec:
  running: true
  dataVolumeTemplates:
  - metadata:
      name: winhd
    spec:
      source:
        blank: {}
      pvc:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 15Gi
  template:
    metadata:
      labels:
        kubevirt.io/domain: win2k12-iso
    spec:
      domain:
        cpu:
          cores: 4
        devices:
          disks:
          - bootOrder: 1
            cdrom:
              bus: sata
            name: cdromiso
          - disk:
              bus: virtio
            name: harddrive
          - cdrom:
              bus: sata
            name: virtiocontainerdisk
        machine:
          type: q35
        resources:
          requests:
            memory: 4G
      volumes:
      - name: cdromiso
        persistentVolumeClaim:
          claimName: iso-win2k12
      - name: harddrive
        dataVolume:
          name: winhd
      - containerDisk:
          image: kubevirt/virtio-container-disk
        name: virtiocontainerdisk
```

**Comment by usrbinkat** [comment]
Given we're seeing hostpath in use here and permissions/locking issues on that hostpath, are you able to install an alternate storage provider/class and test with that also?

**Comment by stefanDeveloper** [comment]
virt-handler output: 

```
{"component":"virt-handler","level":"info","msg":"VMI does not exist | Domain does not exist","pos":"vm.go:1693","timestamp":"2021-12-09T08:43:08.197545Z"}
{"component":"virt-handler","kind":"VirtualMachineInstance","level":"info","msg":"Performing final local cleanup for vmi with uid ","name":"win2k12-iso","namespace":"default","pos":"vm.go:1944","timestamp":"2021-12-09T08:43:08.197653Z","uid":""}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"win2k12-iso","namespace":"default","pos":"vm.go:1979","timestamp":"2021-12-09T08:43:08.197723Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Scheduled | Domain does not exist","name":"win2k12-iso","namespace":"default","pos":"vm.go:1688","timestamp":"2021-12-09T08:43:15.058432Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"info","msg":"Bind mounting container disk at /var/lib/docker/overlay2/2248f35467c249ed90f0cee869dcb655f5aa72d9e59d14d37d7c93fd880fcc1b/merged/disk/downloaded to /var/lib/kubelet/pods/820c0c83-2859-443c-bf73-73b011fb2992/volumes/kubernetes.io~empty-dir/container-disks/disk_2.img","name":"win2k12-iso","namespace":"default","pos":"mount.go:259","timestamp":"2021-12-09T08:43:15.062270Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"info","msg":"mounting kernel artifacts","name":"win2k12-iso","namespace":"default","pos":"mount.go:379","timestamp":"2021-12-09T08:43:15.098788Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"info","msg":"kernel boot not defined - nothing to mount","name":"win2k12-iso","namespace":"default","pos":"mount.go:382","timestamp":"2021-12-09T08:43:15.098867Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","level":"info","msg":"Created tap device: tap0 in PID: 3572261","pos":"common.go:391","timestamp":"2021-12-09T08:43:15.129075Z"}
{"component":"virt-handler","level":"info","msg":"Successfully configured tap device: tap0","pos":"common.go:430","timestamp":"2021-12-09T08:43:15.136277Z"}
{"component":"virt-handler","kind":"","level":"info","msg":"Accepted new notify pipe connection for vmi","name":"win2k12-iso","namespace":"default","pos":"vm.go:335","timestamp":"2021-12-09T08:43:15.403914Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Domain is in state Shutoff reason Unknown","name":"win2k12-iso","namespace":"default","pos":"vm.go:2831","timestamp":"2021-12-09T08:43:15.406164Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Scheduled | Domain status: Shutoff, reason: Failed","name":"win2k12-iso","namespace":"default","pos":"vm.go:1686","timestamp":"2021-12-09T08:46:56.066004Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"info","msg":"mounting kernel artifacts","name":"win2k12-iso","namespace":"default","pos":"mount.go:379","timestamp":"2021-12-09T08:46:56.092408Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"info","msg":"kernel boot not defined - nothing to mount","name":"win2k12-iso","namespace":"default","pos":"mount.go:382","timestamp":"2021-12-09T08:46:56.092493Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","kind":"","level":"error","msg":"Synchronizing the VirtualMachineInstance failed.","name":"win2k12-iso","namespace":"default","pos":"vm.go:1826","reason":"server error. command SyncVMI failed: \"LibvirtError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 2021-12-09T08:46:56.268662Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.4,addr=0x0,drive=libvirt-1-format,id=ua-harddrive,write-cache=on,werror=stop,rerror=stop: Failed to get \\\"write\\\" lock\\nIs another process using the image [/var/run/kubevirt-private/vmi-disks/harddrive/disk.img]?')\"","timestamp":"2021-12-09T08:46:56.275115Z","uid":"94e64c3b-e031-4e8a-acb5-982e52d4e08f"}
{"component":"virt-handler","level":"info","msg":"re-enqueuing VirtualMachineInstance default/win2k12-iso","pos":"vm.go:1479","reason":"server error. command SyncVMI failed: \"LibvirtError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 2021-12-09T08:46:56.268662Z qemu-kvm: -device virtio-blk-pci-non-transitional,bus=pci.4,addr=0x0,drive=libvirt-1-format,id=ua-harddrive,write-cache=on,werror=stop,rerror=stop: Failed to get \\\"write\\\" lock\\nIs another process using the image [/var/run/kubevirt-private/vmi-disks/harddrive/disk.img]?')\"","timestamp":"2021-12-09T08:46:56.275755Z"}
```

I'll test it with alternate storage provider/class and give you feedback

**Comment by stefanDeveloper** [comment]
I figured out what my mistake was, I set the same hostpath as the CDI is using it.
So changing this
```yaml
  hostPath:
    path: "/mnt/data"
```
to 
```yaml
  hostPath:
    path: "/mnt/SOME_OTHER"
```
fixed it. However, now I face the problem that I get an `Can't connect to websocket (504): websocket: bad handshake` when I try to connect via `virtctl vnc win2k12-iso`. The same happens when I try this example `https://kubevirt.io/labs/manifests/vm.yaml` with cirros.
